crates/swc_ecma_transforms_compat/src/es2015/destructuring.rs

```diff
@@ -931,23 +946,35 @@ impl VisitMut for AssignFolder {
                         });
                     }
                     Pat::Assign(pat) => {
-                        let ref_ident = make_ref_ident(self.c, &mut self.vars, None);
+                        let ref_ident = match *pat.right {
+                            Expr::Ident(ref i) if i.span.ctxt() != SyntaxContext::empty() => {
```

Comment:
What is this check?
All passes are invoked after applying `resolver`, so I think all identifier reference expressions will have non-empty `SyntaxContext`.